ckanext/blob_storage/helpers.py

```diff
@@ -87,3 +92,72 @@ def resource_filename(resource):
         url_path = urlparse(resource['url']).path
         return path.basename(url_path)
     return resource['url']
+
+
+def _check_resource_in_dataset(resource_id, dataset_id, context=None):
+    # type: (str, str, OptionalCkanContext) -> bool
+    """Check that a resource exists in the dataset
+    """
+    if context is None:
+        context = get_user_context()
+    try:
+        ds = toolkit.get_action('package_show')(context, {"id": dataset_id})
+        for resource in ds['resources']:
+            if resource['id'] == resource_id:
+                return True
+    except (toolkit.ObjectNotFound, toolkit.NotAuthorized):
+        pass
+
+    return False
+
+
+def activity_resource_show(context, data_dict):
+    activity_id = data_dict.get('activity_id', None)
+    dataset_id = data_dict.get('dataset_id', None)
+    resource_id = data_dict.get('resource_id', None)
+
+    if activity_id and dataset_id and resource_id:
+        return find_activity_resource(activity_id, resource_id, dataset_id, context)[1]
+    else:
+        raise AttributeError("Attribute(s) not found")
+
+
+def find_activity_resource(activity_id, resource_id, dataset_id, context) -> (dict, dict):
+    """check if resource in a release
+    """
+    if activity_id and toolkit.check_ckan_version(min_version='2.9'):
+        try:
+            activity = toolkit.get_action(u'activity_show')(
+                context, {u'id': activity_id, u'include_data': True})
+            activity_dataset = activity['data']['package']
+
+            assert (activity_dataset['name'] == dataset_id) or (activity_dataset['id'] == dataset_id)
+
+            activity_resources = activity_dataset['resources']
+            for r in activity_resources:
+                if r['id'] == resource_id:
+                    resource = r
+                    package = activity_dataset
+                    return package, resource
```

Comment:
I think it's not a good idea to return the tuple `(package, resource)` here.
1. The function name suggest it's finding just the resource
2. In your case you end up with weird usages like: `find_activity_resource(activity_id, resource_id, dataset_id, context)[1]`